api/handlers.go

```diff
@@ -229,12 +234,23 @@ func (api *API) loginGET(_ *database.User, w http.ResponseWriter, req *http.Requ
 // loginPOST starts a user session by issuing a cookie
 func (api *API) loginPOST(_ *database.User, w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
 	// Get the body, we might need to use it several times.
-	body, err := ioutil.ReadAll(io.LimitReader(req.Body, LimitBodySizeSmall))
+	body, err := io.ReadAll(io.LimitReader(req.Body, LimitBodySizeSmall))
 	if err != nil {
 		api.WriteError(w, errors.AddContext(err, "failed to read request body"), http.StatusBadRequest)
 		return
 	}
 
+	// Use custom JWT TTL if defined in the request.
+	var jwtTTL loginTTL
+	err = json.Unmarshal(body, &jwtTTL)
+	if err != nil || jwtTTL.TTL == 0 {
```

Comment:
REQ: we shouldn't silently error here imo. If the Unmarshalling errors it is a user error which should be returned.

Reply:
Makes sense.